guidance/include/guidance/guidance_worker.hpp

```diff
@@ -82,15 +82,20 @@ namespace guidance
             std::vector<std::string> required_plugins;
 
             std::atomic<bool> guidance_activated_;
+
+            std::string vehicle_type;
+
+            void create_guidance_state_machine();
         
         private:
+            GuidanceStateMachineFactory guidance_state_machine_factory;
             // Guidance state machine
-            GuidanceStateMachine gsm;
+            GuidanceStateMachine* gsm;
```

Comment:
Please use smart pointers instead of raw pointers unless there is a justification for it.